packages/studio-plugin/src/orchestrators/ParsingOrchestrator.ts

```diff
@@ -214,7 +214,14 @@ export default class ParsingOrchestrator {
         `The pages directory does not exist, expected directory to be at "${this.paths.pages}".`
       );
     }
-    return createFilenameMapping(this.paths.pages, this.getOrCreatePageFile);
+
+    const excludeReservedPagesJSFiles = (filename: string) =>
+      !(filename.includes("_server") || filename.includes("_client"));
```

Comment:
I can't remember the exact details of our page name validation in `studio-ui`. If a user tried to add a new page with the name `_server` through the UI, will they currently get an error about it being an invalid filename because it starts with `_`?

Reply:
Our validation doesn't seem to catch the `_`. I was able to make a Page Template via Studio  whose page name was `_server`.

Reply:
I see. Then maybe we should specifically block users from creating a page with either of the reserved names because that could cause issues when saving

Reply:
I briefly mentioned this in my commit message. I was planning to wait for the Glob work to add this kind of validation. With Globs, we will also incorporate support for sub-folders. If a Developer restricts Studio users to certain sub-folders of `src/templates`, we'll want to ensure any Pages they create live in one of the approved sub-folders. So, our validation for Page Creation will need a non-trivial overhaul. As part of that overhaul, we'd check for blacklisted .tsx files in the Glob (such as `_client` and `_server`) and make sure new Pages didn't have those names either. 

I can add some quick validation for now though.